From 3d4a9cbfa44c013215b9342392c2a07dd97ca228 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Tue, 28 Oct 2025 11:30:04 +0100
Subject: [PATCH] gcompris: stop linking to Qt6 private targets

fix https://hydra.nixos.org/build/311320100/log/tail
```
CMake Error at src/core/CMakeLists.txt:112 (target_link_libraries):
  Target "gcompris-qt" links to:

    Qt6::CorePrivate

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
---
 pkgs/by-name/gc/gcompris/package.nix | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/pkgs/by-name/gc/gcompris/package.nix b/pkgs/by-name/gc/gcompris/package.nix
index df8f4c6e90f1f..5019329041c07 100644
--- a/pkgs/by-name/gc/gcompris/package.nix
+++ b/pkgs/by-name/gc/gcompris/package.nix
@@ -18,11 +18,17 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-Y23pFov1/WKPrwYYRfGI8sOF0tp/ksSwRJE5zmxtoSo=";
   };
 
-  # fix concatenation of absolute paths like
-  # /nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.1.1/bin/..//nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.1.1/share/gcompris-qt/rcc/core.rcc
   postPatch = ''
+    # fix concatenation of absolute paths like
+    # /nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.1.1/bin/..//nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.1.1/share/gcompris-qt/rcc/core.rcc
     substituteInPlace src/core/config.h.in  --replace-fail \
       "../@_data_dest_dir@" "../share/gcompris-qt"
+
+    # Don't link to private targets, which are not exposed ad
+    substituteInPlace src/core/CMakeLists.txt \
+      --replace-fail "$""{QT_MAJOR}::CorePrivate" "" \
+      --replace-fail "$""{QT_MAJOR}::QuickControls2BasicPrivate" "" \
+      --replace-fail "$""{QT_MAJOR}::WaylandClientPrivate" ""
   '';
 
   cmakeFlags = [
